internal/tableland/tableland.go

```diff
@@ -50,102 +23,27 @@ type TxnReceipt struct {
 	ErrorEventIdx int     `json:"error_event_idx"`
 }
 
-// GetReceiptResponse is a GetTxnReceipt response.
-type GetReceiptResponse struct {
-	Ok      bool        `json:"ok"`
-	Receipt *TxnReceipt `json:"receipt,omitempty"`
-}
-
-// ValidateCreateTableRequest is a ValidateCreateTable request.
-type ValidateCreateTableRequest struct {
-	CreateStatement string `json:"create_statement"`
-}
-
-// ValidateCreateTableResponse is a ValidateCreateTable response.
-type ValidateCreateTableResponse struct {
-	StructureHash string `json:"structure_hash"`
-}
-
-// ValidateWriteQueryRequest is a ValidateWriteQuery request.
-type ValidateWriteQueryRequest struct {
-	Statement string `json:"statement"`
-}
-
-// ValidateWriteQueryResponse is a ValidateWriteQuery response.
-type ValidateWriteQueryResponse struct {
-	TableID string `json:"table_id"`
-}
-
-// SetControllerRequest is a user SetController request.
-type SetControllerRequest struct {
-	Controller string `json:"controller"`
-	TokenID    string `json:"token_id"`
-}
-
-// SetControllerResponse is a RunSQL response.
-type SetControllerResponse struct {
-	Transaction struct {
-		Hash string `json:"hash"`
-	} `json:"tx"`
-}
-
 // SQLRunner defines the run SQL interface of Tableland.
 type SQLRunner interface {
-	RunReadQuery(context.Context, RunReadQueryRequest) (RunReadQueryResponse, error)
+	RunReadQuery(context.Context, string) (interface{}, error)
```

Comment:
Example of a more fluent Go API here, no need for the request/response types.

Reply:
I don't recall from memory, but can this interface be defined in the caller site?

I think it is just the caller trying to strip down the need for the mesa dependency to call `RunReadQuery`; I'm not sure if it makes sense to have that in such a general package as it's here.

Reply:
I'd vote to remove this interface if possible

Reply:
I moved this interface to the caller site.